addons/microsoft_calendar/models/res_users.py

```diff
@@ -50,7 +49,8 @@ def _refresh_microsoft_calendar_token(self):
         }
 
         try:
-            dummy, response, dummy = self.env['microsoft.service']._do_request(MICROSOFT_TOKEN_ENDPOINT, params=data, headers=headers, method='POST', preuri='')
+            endpoint = self.env['microsoft.service']._get_token_endpoint()
+            dummy, response, dummy = self.env['microsoft.service']._do_request(endpoint, params=data, headers=headers, method='POST', preuri='')
```

Comment:
Maybe `do_request` could just call the method internally? Doesn't seem very useful to ask `microsoft.service` for something just to give it to `microsoft.service`?

Reply:
I agree it's a bit overkill to ask `microsoft.service` for something that we're sending back to it... however `_do_request` is also used for other endpoints as well